src/utils/line.ts

```diff
@@ -785,3 +776,45 @@ export const formatOptimisticProposal = (
 
   return proposedPosition;
 };
+
+// TODO: Move code from getLines to create state.categories into this function
+export const formatLineCategories = (
+  lines: { [key: string]: SecuredLine },
+  categories: { [key: string]: string[] }
+): { [key: string]: string[] } => {
+  const validLines = _.filter(lines, (line) => line.status === 'active' && line.end * 1000 > Date.now());
+
+  const highestCreditLines = _.map(
+    _.sortBy(validLines, (line) => Number(BigNumber.from(line.deposit).div(BASE_DECIMALS)))
+      .reverse()
+      .slice(0, 3),
+    'id'
+  );
+
+  const highestCreditLinesCategory = { 'market:featured.highest-credit': highestCreditLines };
+
+  const highestRevenueLines = _.map(
+    _.sortBy(validLines, (line) => Number(BigNumber.from(line!.spigot!.revenueValue).div(BASE_DECIMALS)))
+      .reverse()
+      .slice(0, 3),
+    'id'
+  );
+
+  const highestRevenueLinesCategory = { 'market:featured.highest-revenue': highestRevenueLines };
+
+  const newCreditLines = _.map(
+    _.sortBy(validLines, (line) => line.start),
+    'id'
+  );
+
+  const newCreditLinesCategory = { 'market:featured.newest': newCreditLines };
+
+  const updatedCategories = {
+    ...categories,
```

Comment:
you arent really using the `category` param. You are hardcoding the category names already. Pretty sure u can delete

Reply:
i guess u technically do use it back in the action but i'd prefer if u didnt pass it in just to directly return it to reduce complexity. Can destructure `categories` and `formattedCategories` into state inside action

Reply:
100% agree.

changes made!